controllers/component_build_controller_common.go

```diff
@@ -92,6 +97,33 @@ func getGitProvider(component appstudiov1alpha1.Component) (string, error) {
 	return gitProvider, err
 }
 
+// GetPipelineFromComponentAnnotation parses pipeline annotation on component and returns pipeline
+func GetPipelineFromComponentAnnotation(ctx context.Context, component *appstudiov1alpha1.Component) (*tektonapi.PipelineRef, error) {
+	log := ctrllog.FromContext(ctx)
+
+	buildPipeline, err := readBuildPipelineAnnotation(component)
+	if err != nil {
+		return nil, err
+	}
+
+	if buildPipeline.Bundle != "" {
+		log.Info(fmt.Sprintf("Will use default pipeline from annotation %s : name: %s; bundle: %s", defaultBuildPipelineAnnotation, buildPipeline.Name, buildPipeline.Bundle))
```

Comment:
I think we can log this in the main reconcile, so we can avoid passing `ctx` here and in `generatePaCPipelineRunForComponent`

Reply:
done